x-pack/plugins/observability_solution/observability_logs_explorer/public/components/logs_explorer_top_nav_menu.tsx

```diff
@@ -30,13 +30,15 @@ import { AlertsPopover } from './alerts_popover';
 
 export const LogsExplorerTopNavMenu = () => {
   const {
-    services: { serverless },
+    services: { chrome },
   } = useKibanaContextForPlugin();
 
-  return Boolean(serverless) ? <ServerlessTopNav /> : <StatefulTopNav />;
+  const chromeStyle = useObservable(useMemo(() => chrome.getChromeStyle$(), [chrome]));
```

Comment:
**question:** is this memoization here to prevent some rerenders or because the `chrome.getChromeStyle$` invocation is expensive?

Reply:
I memoized it here because every call to `chrome.getChromeStyle$` creates a new observable (because it's ["cold"](https://rxjs.dev/guide/glossary-and-semantics#cold)). Without the memoization every re-render would spawn a new producer leading to a lot of memory churn. Does that make sense?

Reply:
Totally thanks, I was expecting the `getChromeStyle$` to return the same referenced observable, but in this case the memo makes sense 👌

Reply:
It seems like something we might want to fix on our side though... I wouldn't expect consumers to have to use `useMemo`.

Reply:
Yes, I could change this. I just modeled this after all the other observable-returning methods in the chrome service for now.

Reply:
We use it without memo here: https://github.com/elastic/kibana/blob/main/packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx#L355

Is it the `pipe` (https://github.com/elastic/kibana/blob/main/packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx#L534) that creates every time a new Observable?

Reply:
That usage references the `BehaviorSubject` directly, so the reference stays constant. But `getChromeStyle$()` creates a new subscription chain on every call, so the observable identity that `useObservable` subscribes to changes on every render.

Reply:
Ok thanks for explaining 👍 I can do it, or if you want to go ahead you can change in the ChromeService

```ts
const chromeStyleSub$ = new BehaviorSubject<ChromeStyle | undefined>(undefined);
const chromeStyle$ = chromeStyleSub$.pipe(
  filter((style): style is ChromeStyle => style !== undefined),
  takeUntil(this.stop$)
);
```

and then replace `chromeStyleSub$` with `chromeStyleSub$` when the value is set to the Subject.

That should fix this issue, right?

Reply:
yes, that should do it. I can include it in this PR